src/features/withdraw/components/WithdrawForm/hooks/useBlockchainSelectItems.ts

```
@@ -0,0 +1,91 @@
+import {
+  addNep141ToAddress,
+  compareAmounts,
+  filterOutPoaBridgeTokens,
+  getTokenAssetIdsWithoutNep141,
+} from "../../../../../utils/tokenUtils"
+import { allBlockchains } from "../constants"
+
+export const useBlockchainSelectItems = (
+  token: SwappableToken,
+  tokensUsdPriceData?: TokenUsdPriceData,
+  tokenInBalance?: TokenValue
+) => {
+  const availableBlockchains = isBaseToken(token)
+    ? { [token.chainName]: token.defuseAssetId }
+    : token.groupedTokens.reduce((acc: { [key: string]: string }, curr) => {
+        acc[curr.chainName] = curr.defuseAssetId
+        return acc
+      }, {})
+
+  const onlyPoaToken = filterOutPoaBridgeTokens(token)
+  const addresses = onlyPoaToken
+    ? getTokenAssetIdsWithoutNep141(onlyPoaToken)
+    : []
+  const { amount: userBalance, decimals: userBalanceDecimals } =
+    tokenInBalance || { amount: 0n, decimals: 1 }
+  let showHotBalances = userBalance > 0
+
+  const { data } = useTokenBalancesQueryOptions(addresses, showHotBalances)
+
+  let anyHotBalanceIsLessThanUserBalance = false
+
+  const balances =
+    data?.reduce((acc: { [key: string]: TokenBalances }, curr) => {
+      if (
+        compareAmounts(
+          { amount: userBalance, decimals: userBalanceDecimals },
+          { amount: BigInt(curr.vaultBalance), decimals: curr.decimals }
+        ) > 0
+      ) {
+        // we should show in case user's balance is MORE than any of the HOT chain balances
+        anyHotBalanceIsLessThanUserBalance = true
+      }
+
+      acc[addNep141ToAddress(curr.nearAddress)] = curr
+      return acc
+    }, {}) || {}
+
+  showHotBalances = showHotBalances && anyHotBalanceIsLessThanUserBalance
+
+  const blockchainSelectItems = Object.fromEntries(
+    allBlockchains
+      .filter((blockchain) => availableBlockchains[blockchain.value])
+      .map((a) => {
+        const address = availableBlockchains[a.value]
+        let hotBalance = null
+
+        if (address != null) {
```

Also it should be always defined. It's better to just throw an error if it doesn't.